web/Services/AdminService.cs

```diff
@@ -44,11 +57,43 @@ public void CancelAdminAccessRequest(BricksAndHeartsUser user)
         userRecord.HasRequestedAdmin = false;
         _dbContext.SaveChanges();
     }
+    
+    public void ApproveAdminAccessRequest(int userId)
+    {
+        var userToAdmin = _dbContext.Users.Single(u => u.Id == userId);
+        
+        userToAdmin.IsAdmin = true;
+        userToAdmin.HasRequestedAdmin = false;
+        _dbContext.SaveChanges();
+        _logger.LogInformation("Admin request approved");
+    }
+    
+    public void RejectAdminAccessRequest(int userId)
+    {
+        var userToAdmin = _dbContext.Users.Single(u => u.Id == userId);
+        
+        userToAdmin.HasRequestedAdmin = false;
+        _dbContext.SaveChanges();
+        _logger.LogInformation("Admin request rejected");
+    }
 
     public async Task<(List<UserDbModel> CurrentAdmins, List<UserDbModel> PendingAdmins)> GetAdminLists()
     {
         return (await GetCurrentAdmins(), await GetPendingAdmins());
     }
+    
+    private async Task<List<UserDbModel>> GetCurrentAdmins()
+    {
+        var currentAdmins = await _dbContext.Users.Where(u => u.IsAdmin == true).ToListAsync();
+        return currentAdmins;

```

Comment:
Not your code, but anyway
```suggestion
        return await _dbContext.Users.Where(u => u.IsAdmin == true).ToListAsync();
```